terraform/cloud-platform-components/README.md

```diff
@@ -82,6 +83,114 @@ Metrics-server allows us to perform resource queries against the cluster. Comman
+	  - Labelled with openpolicyagent.org/policy=rego.
+
+When a policy has been successfully loaded into OPA, the openpolicyagent.org/policy-status annotation is set to {"status": "ok"}.
+If loading fails for some reason (e.g., because of a parse error), the openpolicyagent.org/policy-status annotation is set to {"status": "error", "error": ...} where the error field contains details about the failure.
+
+IMP NOTE:  Apply a ConfigMap that contains the main OPA policy and default response. This policy is used as an entry-point for policy evaluations and returns allowed:true if policies are not matched to inbound data.
+
+Please see the following documentation by opa on how to do this https://www.openpolicyagent.org/docs/kubernetes-admission-control.html 
+
+ ### How to write Policies
+
+OPA is purpose built for reasoning about information represented in structured documents. The data that our service and its users publish can be inspected and transformed using OPA’s native query language Rego.
+
+Rego was inspired by Datalog, which is a well understood, decades old query language. Rego extends Datalog to support structured document models such as JSON.
+Rego queries are assertions on data stored in OPA. These queries can be used to define policies that enumerate instances of data that violate the expected state of the system.
+
+In Rego, policies are defined inside modules. Modules consist of:
+
+    - Exactly one Package declaration.
+    - Zero or more Import statements.
+    - Zero or more Rule definitions.
+
+Please see the following documentation by opa on how to write Policies https://www.openpolicyagent.org/docs/how-do-i-write-policies.html
+
+ ### kube-mgmt-Replicate
+
+Additionally, kube-mgmt is configured to periodically pull resource metadata that might be needed by the opa service to correctly evaluate API server events, in the case that the event does not contain all the needed data for the logic defined in the evaluation policy. The resources that kube-mgmt scans are configured in templates/opa/values.yaml.tpl under mgmt/replicate:
+
+  ```yaml
+    group/version/resource			  # namespace scoped
+    group/version/resource        # cluster scoped
+  ```
+    <replicate-path> is configurable (via path) and defaults to kubernetes.
+
+The example below would replicate namespaces and ingress into OPA:
+
+```yaml
+    cluster:
+      - "v1/namespaces"
+    namespace:
+      - "extensions/v1beta1/ingresses"
+    path: kubernetes
+```
+
+NOTE: IF we use replicate: remember to update the RBAC rules as below to allow permissions to replicate these things
+
+ ### opa-rbac-resources
+
+RBAC provides a simple, coarse-grained way of granting permissions by groupings. Determining whether to allow requests under RBAC involves identifying whether the caller has been associated with a role that grants permission to the perform the operation. This can be configured in templates/opa/values.yaml.tpl under rbac:
+
```

Comment:
I suggest adding a description of what the example does, i.e. something like;

> The following RBAC definition enables listing, viewing and watching namespaces, and listing viewing watching and patching ingresses....

Reply:
Removed the RBAC section considering the comments from Mitch